test/pipelinefinally_test.go

```diff
@@ -110,11 +163,15 @@ func TestPipelineLevelFinally_OneDAGTaskFailed_Failure(t *testing.T) {
 			if !isSkipped(t, n, taskrunItem.Status.Conditions) {
 				t.Fatalf("dag task %s should have skipped due to condition failure", n)
 			}
+		case n == "dagtask4":
```

Comment:
nitpick but maybe it would make sense to also assert on the length of taskrunList? apologies since this test was already like this, but it seems like it would currently pass if extra taskruns were returned

Reply:
sure I will add it explicitly, it is covered by the default:

https://github.com/tektoncd/pipeline/blob/705e97146fea0e6a7f55e3b06f46e317a9c12659/test/pipelinefinally_test.go#L118-L119

`default` would raise error if it finds any missing or extra taskRuns, I will add explicit count assertion.

Reply:
will update this error message as well since its not clear.

Reply:
https://github.com/tektoncd/pipeline/pull/3242/files#diff-7815ee756d1326fe39c09f4deb5c7c36aa4a8f14b4319d5bf8ac73a1929a3d10R168